tests/rust/tcp-test/accept/mod.rs

```diff
@@ -0,0 +1,193 @@
+    println!("{}", stringify!(accept_listening_socket));
+
+    // Create an accepting socket.
+    let sockqd: QDesc = libos.socket(AF_INET, SOCK_STREAM, 0)?;
+    libos.bind(sockqd, local.to_owned())?;
+    libos.listen(sockqd, 16)?;
+
+    // Succeed to accept() connections.
+    let qt: QToken = libos.accept(sockqd)?;
+
+    // Poll once to ensure that the accept() co-routine runs.
+    match libos.wait(qt, Some(Duration::from_micros(0))) {
+        Err(e) if e.errno == libc::ETIMEDOUT => {},
+        Ok(_) => anyhow::bail!("wait() should not succeed"),
+        Err(_) => anyhow::bail!("wait() should timeout"),
+    }
+
+    // Succeed to close socket.
+    libos.close(sockqd)?;
+
+    Ok(())
+}
+
+/// Attempts to accept connections on a TCP socket that is connecting.
+fn accept_connecting_socket(libos: &mut LibOS, remote: &SocketAddrV4) -> Result<()> {
+    println!("{}", stringify!(accept_connecting_socket));
+
+    // Create a connecting socket.
+    let sockqd: QDesc = libos.socket(AF_INET, SOCK_STREAM, 0)?;
+    let qt: QToken = libos.connect(sockqd, remote.to_owned())?;
+
+    // Poll once to ensure that the connect() co-routine runs.
+    match libos.wait(qt, Some(Duration::from_micros(0))) {
+        Err(e) if e.errno == libc::ETIMEDOUT => {},
+        Ok(_) => anyhow::bail!("wait() should not succeed"),
+        Err(_) => anyhow::bail!("wait() should timeout"),
+    }
+
+    // Fail to accept() connections.
+    let e: Fail = libos
+        .accept(sockqd)
+        .expect_err("accept() connections on a socket that is connecting should fail");
+
+    // Sanity check error code.
+    assert_eq!(e.errno, libc::EINVAL, "accept() failed with {}", e.cause);
+
+    // Succeed to close socket.
+    libos.close(sockqd)?;
+
+    Ok(())
+}
+
+/// Attempts to accept connections on a TCP socket that is already accepting connections.
+fn accept_accepting_socket(libos: &mut LibOS, local: &SocketAddrV4) -> Result<()> {
+    println!("{}", stringify!(accept_accepting_socket));
+
+    // Create an accepting socket.
+    let sockqd: QDesc = libos.socket(AF_INET, SOCK_STREAM, 0)?;
+    libos.bind(sockqd, local.to_owned())?;
+    libos.listen(sockqd, 16)?;
+    let qt: QToken = libos.accept(sockqd)?;
+
+    // Poll once to ensure that the accept() co-routine runs.
+    match libos.wait(qt, Some(Duration::from_micros(0))) {
+        Err(e) if e.errno == libc::ETIMEDOUT => {},
+        Ok(_) => anyhow::bail!("wait() should not succeed"),
+        Err(_) => anyhow::bail!("wait() should timeout"),
+    }
+
+    // Fail to accept() connections.
+    let e: Fail = libos
+        .accept(sockqd)
+        .expect_err("accept() connections on a socket that is accepting connections should fail");
+
+    // Sanity check error code.
+    assert_eq!(e.errno, libc::EINPROGRESS, "accept() failed with {}", e.cause);
+
+    // Succeed to close socket.
+    libos.close(sockqd)?;
+
+    Ok(())
+}
```

Comment:
We might want to add accept_after_close

Reply:
Sure I added this extra test to issue https://github.com/demikernel/demikernel/issues/549.